### PR TITLE
Login : card, form, responsive

### PR DESCRIPTION
**Pull request by Jordanras96**
installation of card shadcn ui. integration of "Login card", 90% responsive. integration of card logo and image.

Need a code review to see if everything's ok so far.

TO DO (for tomorrow or for those who are free):
for responsive, we still need to define the minimum size of inputs, buttons, spans and images to prevent them from being too small before the breakpoint on lg to md and for sm to smaller.

### CODE REVIEW

File: app/routes/_auth+/login.tsx [hunk 1]
```diff
@@ -126,24 +156,32 @@ export default function LoginPage() {
...
-                }}
-              >
-                Sign up
-              </Link>
+            <button
+              type="submit"
+              className="w-full rounded bg-primary px-4 py-2 text-white hover:bg-primary/80 focus:bg-blue-400"
+            >
+              Connexion
+            </button>
```
Comment (fenny-mdg) [inline app/routes/_auth+/login.tsx]: Utiliser le Button de Shadcn

File: app/routes/_auth+/login.tsx [hunk 2]
```diff
@@ -80,41 +85,66 @@ export default function LoginPage() {
...
+            <div id={form.errorId} className="pt-1 text-red-700">
+              {form.errors}
+            </div>
+            <Field
+              labelProps={{ children: "Email" }}
+              inputProps={{
+                ...getInputProps(fields.email, {
+                  type: "email",
+                }),
+                placeholder: "user@gmail.com",
```
Comment (fenny-mdg) [inline app/routes/_auth+/login.tsx]: utiliser un placeholder plus générique comme: user@example.com

File: app/routes/_auth+/login.tsx [hunk 3]
```diff
@@ -80,41 +85,66 @@ export default function LoginPage() {
...
+    <div className="bg-[#DFDACF] size-full flex justify-center items-center overflow-x-hidden overscroll-none ">
+      <img
+        src={LineImg}
+        alt="ligne"
+        className="fixed left-0 right-full z-10 min-h-full hidden lg:block"
+      />
+      <Card
+        className={cn(
+          "max-w-7xl container mx-auto bg-white p-5 flex shadow-md rounded-2xl h-screen lg:w-1/2 lg:h-auto ",
+        )}
```
Comment (fenny-mdg) [inline app/routes/_auth+/login.tsx]: enlever cn

File: app/routes/_auth+/login.tsx [hunk 4]
```diff
@@ -10,13 +10,18 @@ import { Form, Link, useActionData, useSearchParams } from "@remix-run/react";
 import { useEffect, useRef } from "react";
 import { z } from "zod";
 
+import { Card } from "@/components/ui/card";
+import { cn } from "@/lib/utils";
 import { Field } from "~/components/forms";
 import { createUserSession, getUserId } from "~/utils/session.server";
 import { verifyLogin } from "~/utils/user.server";
 
+import LineImg from "../../../public/assets/brand/LineImg.svg";
```
Comment (fenny-mdg) [inline app/routes/_auth+/login.tsx]: renommer en lineImg

File: app/root.tsx [hunk 5]
```diff
@@ -23,6 +37,10 @@ export const loader = async ({ request }: LoaderFunctionArgs) => {
 };
 
 export default function App() {
+  const pathname = useLocation().pathname;
+
+  const shouldHideOverflow =
+    pathname === "/login" || pathname === "/join" || pathname === "/logout";
```
Comment (fenny-mdg) [inline app/root.tsx]: Trouvez une autre alternative

File: app/components/forms.tsx [hunk 6]
```diff
@@ -5,6 +5,14 @@ import { Label } from "@/components/ui/label";
 
 export type ListOfErrors = (string | null | undefined)[] | null | undefined;
 
+interface FieldProps {
```
Comment (fenny-mdg) [inline app/components/forms.tsx]: Utiliser plutôt type

**Review by fenny-mdg — CHANGES_REQUESTED** [review]
Voir les commentaires

File: app/components/forms.tsx (hunk 6 above)
Comment (Jordanras96) [reply app/components/forms.tsx]: j'ai l'erreur : "Use an `interface` instead of a `type`." Peux tu faire des modifications côté lint s'il te plait

Comment (fenny-mdg) [reply app/components/forms.tsx]: c'est fait

**Review by fenny-mdg — APPROVED** [review]

**Pull request merged**